docs/proposals/PROPOSAL_A.md

```diff
@@ -0,0 +1,115 @@
+# Proposal A - FreeBSD Jails
+
+This proposal recommends changes to describe the requirements for mapping an OCI
+runtime container to a corresponding FreeBSD jail.
+
+## Modifications
+
+This suggests adding an object to the [FreeBSD-specific section](https://github.com/opencontainers/runtime-spec/blob/main/config.md#platform-specific-configuration) of the [container configuration](https://github.com/opencontainers/runtime-spec/blob/main/config.md) to describe the required parameters for the jail.
+
+## Jail Configuration
+
+Jail parameters to use for the container's jail.
+
+**`jail`** _(object, OPTIONAL)_ jail parameters for this container.
+
+The following parameters can be specified for for the container jail:
+
+- **`parent`** _(string, OPTIONAL)_ - parent jail.
+    The value is the name of a jail which should be this container's parent (defaults to none).
+- **`host`** _(string, OPTIONAL)_ - allow overriding hostname, domainname, hostuuid and hostid.
+    The value can be "new" which allows these values to be overridding in the container or "inherit" to use the host values. If set to "new", the values for hostname and domainname are taken from the base config, if present.
+- **`ip4`** _(string, OPTIONAL)_ - control the availability of IPv4 addresses.
+    The value can be "new" which allows the addresses listed in **`ip4Addr`** to be used, "inherit" which allows all addresses in the jail's vnet or "disable" to stop use of IPv4 entirely.
+- **`ip4Addr`** _(array of string, OPTIONAL)_ - list of IPv4 addresses usable by the jail
+- **`ip6`** _(string, OPTIONAL)_ - control the availability of IPv6 addresses.
+    The value can be "new" which allows the addresses listed in **`ip6Addr`** to be used, "inherit" which allows all addresses in the jail's vnet or "disable" to stop use of IPv6 entirely.
+- **`ip6Addr`** _(array of string, OPTIONAL)_ - list of IPv6 addresses usable by the jail
+- **`vnet`** _(string, OPTIONAL)_ - control the vnet used for this jail.
+    The value can be "new" which causes a new vnet to be created for the jail or "inherit" which shares the vnet for the parent (or host if there is no parent).
+- **`enforceStatfs`** _(integer, OPTIONAL)_ - control visibility of mounts in the jail.
```

Comment:
Should we use magic numbers (0, 1, 2) or define some enums here?

Reply:
I was using strings here mostly because I don't know how to do enum in json. We could use the magic numbers which slightly simplifies the runtime implementation, assuming the numbers match what the kernel jail API expects. I slightly prefer strings to numbers since it makes the config json more readable.